### DISCUSSION TITLE
What can cause an error login?

### DISCUSSION THREAD

**Discussion by loldjy123** [shashi278/social-auth-kivy#17] [Q&A] [answered]
Hello! I trying to put only google authentication in my app , but i got a error login every time i try to log in, the func error_listener it's called each time but it's working properly in my computer , what can cause this issue, how can i fixed that ?

**Comment by shashi278** [comment]
Are you facing the error in the android application? In any case, looking at the logs might be helpful.

**Comment by shashi278** [comment]
Hi, no flask is only used on Desktop and not on Android so it's not causing
the error. As I said, you need to look at the adb logs to see what's going
on there.

**Comment by loldjy123** [comment]
Hi dear I'm so sorry for boring you, I was unable to adb logs in my computer but now i finally able to do so,  I think i get something in the output from the logcat

python  : [INFO   ] [KivyAuth    ] Error signing in using Google. JVM exception occurred: 10:  com.google.android.gms.common.api.ApiException

Everything else seems fine . Do you have any idea about that ?

more info fro the output : 

01-28 02:34:15.979 26752 26752 W SignIn  : [Activity,SignInChimeraActivity] Failed to record the consent.
01-28 02:34:16.340  2869  2869 I GoogleInputMethodServic: GoogleInputMethodService.onFinishInput():3316
01-28 02:34:16.344  2869  2869 I GoogleInputMethodServic: GoogleInputMethodService.updateDeviceLockedStatus():2181 checkRepeatedly = false, unlocked = true
01-28 02:34:16.345  2869  2869 I GoogleInputMethodServic: GoogleInputMethodService.onStartInput():1989
01-28 02:34:16.345  2869  2869 I GoogleInputMethodServic: GoogleInputMethodService.updateDeviceLockedStatus():2181 checkRepeatedly = true, unlocked = true
01-28 02:34:16.367 26752 26752 W AutoManageHelper: Unresolved error while connecting client. Stopping auto-manage.
01-28 02:34:16.390  2869  2869 I GoogleInputMethodServic: GoogleInputMethodService.onFinishInput():3316
01-28 02:34:16.391  2869  2869 I GoogleInputMethodServic: GoogleInputMethodService.updateDeviceLockedStatus():2181 checkRepeatedly = false, unlocked = true
01-28 02:34:16.391  2869  2869 I GoogleInputMethodServic: GoogleInputMethodService.onStartInput():1989
01-28 02:34:16.392  2869  2869 I GoogleInputMethodServic: GoogleInputMethodService.updateDeviceLockedStatus():2181 checkRepeatedly = true, unlocked = true
01-28 02:34:16.537 27257 27257 V PythonActivity: onResume()
01-28 02:34:16.537 27257 27257 V SDL     : onResume()
01-28 02:34:16.538  2869  2869 I GoogleInputMethodServic: GoogleInputMethodService.onFinishInput():3316
01-28 02:34:16.544  2869  2869 I GoogleInputMethodServic: GoogleInputMethodService.updateDeviceLockedStatus():2181 checkRepeatedly = false, unlocked = true
01-28 02:34:16.544  2869  2869 I GoogleInputMethodServic: GoogleInputMethodService.onStartInput():1989
01-28 02:34:16.545 27257 27257 I python  : [INFO   ] [KivyAuth    ] google_activity_listener called.
01-28 02:34:16.545  2869  2869 I GoogleInputMethodServic: GoogleInputMethodService.updateDeviceLockedStatus():2181 checkRepeatedly = true, unlocked = true
01-28 02:34:16.604 27257 27257 I python  : [INFO   ] [KivyAuth    ] Error signing in using Google. JVM exception occurred: 10:  com.google.android.gms.common.api.ApiException
01-28 02:34:16.618 27257 27257 E libEGL  : call to OpenGL ES API with no current context (logged once per thread)

**Reply by bphd** [reply]
E/Finsky  (): [173] uks.a(849): Encountered exception whilst fetching from GMS, deferring to Finsky: com.google.android.gms.common.api.ApiException: 17: API: IntegrityApiDisplayListener.API is not available on this device. Connection failed with: ConnectionResult{statusCode=API_DISABLED, resolution=null, message=null}

**Comment by shashi278** [comment]
Yes, the issue is known. It comes up when the SHA key on your system and the one you used to create API key in google console, are different.

**Reply by shashi278** [reply]
@loldjy123 Make sure to follow the [prerequisites for google login](https://github.com/shashi278/social-auth-kivy/blob/master/docs/prerequisites.md#for-google-login)

**Comment by loldjy123** [comment]
Thank you sir, i will follow the instructions and I’ll keep you update. Thank you again for you help

**Comment by loldjy123** [comment]
Hello Dear I'm still experencing the same issue, so i have a few question to try to figure out what;s wrong.

What is the correct package name to put on googe cloud console ? Is that the package.name from buildozer.spec file or the package domaine + package name.

What is the best way to find SHA-1 in cmd ? reason why i'm asking it's because they are 2 way that i look for it,  and I got 2 different result.
the first way I get the SHA is : 
keytool -keystore newkey.keystore -list -v
second way is:
keytool -exportcert -alias androiddebugkey -keystore "C:\Users\%USERPROFILE%\.android\debug.keystore" | "C:\Users\%PATH%\sha\bin\openssl" sha1 -binary | "C:\Users\%PATH%\sha\bin\openssl" base64

what do you think is the best one?

Can you please give me some advise dear? Thank you

**Reply by usMAEC** [reply]
any luck? I am having the same issue

**Comment by usMAEC** [comment]
Works on desktop with both Client_ID and Client_Secret but not on mobile with the android oAuth2.0

**Reply by usMAEC** [reply]
any thing we are missing here as the issue for me is identical

**Reply by usMAEC** [reply]
@shashi278

**Comment by loldjy123** [comment]
@usMAEC  it's working for me now, if you target android you don't have to put Client_id and secret_id, you can just write initialize google function like initialize_google(self.after_login, self.error_listener).

one last thing, when you are creating the google OAuth, you have to put SHA-1 fingerprint from the system that you are building you app, for example if you using windows and build the app on  Ubuntu system, Don't take sha-1 from you windows, you have to take it from the ubuntu system.

**Reply by luisadrianpuga** [reply]
Thank you. I am compiling this to a keystore folder in my home directory. Do I copy and paste the key to my app folder(also in my home directory ) or should I be making these inside the app folder? (ubuntu)

**Reply by loldjy123** [reply]
i think you can do it inside the app folder in the Ubuntu system, you can use that command

keytool -list -v -keystore ~/.android/debug.keystore -alias androiddebugkey -storepass android -keypass android

it worked for me

**Reply by AustinHellerRepo** [reply]
I fixed this by having both a desktop and android OAuth client, supplying the desktop client ID and secret when testing locally and supplying None for both fields when publishing the apk. Using "from kivy import platform" to do the check to see "if platform == 'android':" at the point where I initialize_google.

**Reply by shashi278** [reply]
No need to check explicitly as the library does that internally. The client ID and secret is used only when you're compiling it for desktop, they've no use when you're compiling for Android. It's been designed in such a way that one code can be used to compile for both the platforms.

**Reply by AustinHellerRepo** [reply]
I just tested it and it looks like I still need the desktop credential (for local testing) and the android credential (for Android devices) setup on Google Cloud Platform, but that for Android I do not need to care about supplying the client ID and secret (so I do not need an if-statement, just passing in the desktop client ID and secret to be unused). Thank you for clarifying that.
My original issue stemmed from not having both a desktop credential and an Android credential setup in Google Cloud Platform.

**Reply by shashi278** [reply]
Glad I could clarify that. I know this thing should be there in the doc because it creates some confusion.
So it's like:
* if you're building only for desktop, you'd need to pass the credentials in the initialize thing. 
* if you're building only for Android, no need to pass credentials in the initialize thing. Even if you pass credentials, they'll be ignored. Just make sure you've it setup in Google cloud platform, or Facebook, or whatever you're targeting. 
* if you're building for both, cross platform, then you'd need to pass the credentials as well as have it setup in the cloud consoles.
* The same initialize function works for all cases without needing explicit platform conditions.

**Comment by shashi278** [comment, marked as answer]
Make sure you enter correct package name and SHA-1 fingerprint in Google cloud console while creating OAuth client. Those are the only mismatch fields that leads to this issue.

**Reply by luisadrianpuga** [reply]
@shashi278 Would this be the the package name we rename in the bulldozer spec or the package name of the apk in the bin after it compiles? Thank you for reading this far.

**Reply by shashi278** [reply]
@luisadrianpuga They're same. Your package name doesn't change upon compilation.

**Reply by luisadrianpuga** [reply]
Thank you, @shashi278 .

**Reply by shashi278** [reply]
Also just a heads up, the complete package name for your Android application is
`package.domain`.`package.name` from buildozer.spec.

For example if,
package.name = login
package.domain = com.test

Entire package name would be: com.test.login

**Reply by luisadrianpuga** [reply]
Thank you for the heads up, @shashi278. I believe im on the right track now. Cheers.

**Reply by jmangefrida** [reply]
This right here saved me.  I'd been trying to get this to work for days.  It makes sense that the package domain would need to be included, but I just never thought of it.

**Reply by CatchHell** [reply]
When using App Signing by Google Play and Firebase, you need to add the SHA-1 fingerprint of the App signing certificate (found on Google Play Console/ Release Management/ App signing certificate) to the Firebase Console/ Settings/ SHA certificate fingerprints

Updated location for the SHAs: Google Play Console > Release > Setup > App integrity

From: https://stackoverflow.com/questions/47619229/google-sign-in-failed-com-google-android-gms-common-api-apiexception-10

**Reply by shashi278** [reply]
Thanks for the input @CatchHell . I'm sure it'll be helpful for the folks trying this out!
